extensions/poetry/main.ts

```diff
@@ -0,0 +1,53 @@
+import { PhylumApi } from "phylum"
+import { parse } from "https://deno.land/std/flags/mod.ts"
+
+// Add packages to the lockfile, and upload it to Phylum for analysis.
+// Return `true` if the packages pass the project's thresholds.
+// Return `false` if the packages don't pass the thresholds, or the processing
+// job is not finished.
+async function poetryCheck(args: string[]) {
+    console.log("Updating package lock…")
+
+    let process = Deno.run({ cmd: ["poetry", "add", '--lock', ...args] })
```

Comment:
```suggestion
    let process = Deno.run({ cmd: ["poetry", "add", ...args] })
```
It would be surprising to a user to run `poetry add`, expecting to have a specific package added to _both_ the `pyproject.toml` and `poetry.lock` files, only to have this `--lock` argument added...which says to update the `poetry.lock` file and not the `pyproject.toml` file.

Reply:
We need the `--lock` argument because we don't actually want to install the package until we have validated it passes thresholds.

I probably missed adding `poetry install`, or an equivalent method, in case of success. What would be the appropriate action to take, in this case? Is `poetry install` ok, even though it installs _all_ packages in the lockfile?

Reply:
Yes, `poetry install` would work here. It updates the active/current poetry virtual environment with the packages from the `poetry.lock` file. If an environment already exists, only the new/different packages will be added/modified.